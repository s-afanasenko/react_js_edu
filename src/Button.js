import React from 'react';

class Button extends React.Component {
	render() {
		const {
			children, 
			clickHundler, 
			disabled,
			variant,
			size
		} = this.props;

		return (
			<button 
				className={`btn btn-${variant} btn-${size}`}
				onClick={clickHundler} 
				disabled={disabled}>
				{children}
			</button>			
		);
	}
}

Button.defaultProps = {
	variant: 'primary'
}

export default Button;
