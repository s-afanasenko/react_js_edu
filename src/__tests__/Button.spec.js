import "@testing-library/jest-dom/extend-expect";
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../Button';

describe('<Button />', () => {
	it('should render with default props and match snapshot', () => {
		const { container } = render(
			<Button>Click Me!</Button>
		);

		expect(container).toMatchSnapshot(); 
	});

	it('should execute callback on click', () => {
		const callback = jest.fn();
		const { getByText } = render(
			<Button clickHundler={callback}>Click Me!</Button>
		);

		fireEvent.click(getByText('Click Me!'));

		expect(callback).toBeCalledTimes(1);
	});

	it('should be disabled with disabled="true"', () => {
		const { getByText } = render(
			<Button disabled={true}>Click Me!</Button>
		);

		expect(getByText('Click Me!')).toBeDisabled();
	});
});
