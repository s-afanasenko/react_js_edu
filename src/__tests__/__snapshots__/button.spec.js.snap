// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render with default props and match snapshot 1`] = `
<div>
  <button
    class="btn btn-primary btn-"
  >
    Click Me!
  </button>
</div>
`;
